apps/prairielearn/src/sprocs/sync_course_instances.sql

```diff
@@ -144,52 +144,56 @@ BEGIN
         AND dest.course_id = syncing_course_id
         AND (src.errors IS NULL OR src.errors = '');
 
-    -- Now, loop over all valid course instances and sync access rules for them
-    FOR valid_course_instance IN (
+    -- Now, sync access rules for all valid course instances.
+    WITH valid_course_instances AS (
         SELECT short_name, data
         FROM disk_course_instances AS src
         WHERE (src.errors IS NULL OR src.errors = '')
-    ) LOOP
-        SELECT ci.id, ci.display_timezone
-        INTO STRICT syncing_course_instance_id, course_instance_timezone
+    ), synced_course_instances AS (
+        SELECT
+            ci.id,
+            ci.display_timezone,
+            vci.data
         FROM course_instances AS ci
+        JOIN valid_course_instances AS vci ON (vci.short_name = ci.short_name)
         WHERE
-            ci.short_name = valid_course_instance.short_name
+            ci.course_id = syncing_course_id
             AND ci.deleted_at IS NULL
-            AND ci.course_id = syncing_course_id;
-
+    ), inserted_access_rules AS (
         INSERT INTO course_instance_access_rules (
             course_instance_id,
             number,
             uids,
             start_date,
             end_date,
             institution
-        ) SELECT
-            syncing_course_instance_id,
+        )
+        SELECT
+            ci.id AS course_instance_id,
             number,
             CASE
                 WHEN access_rule->'uids' = null::JSONB THEN NULL
                 ELSE jsonb_array_to_text_array(access_rule->'uids')
-            END,
-            input_date(access_rule->>'start_date', course_instance_timezone),
-            input_date(access_rule->>'end_date', course_instance_timezone),
-            access_rule->>'institution'
+            END AS uids,
+            input_date(access_rule->>'start_date', ci.display_timezone) AS start_date,
+            input_date(access_rule->>'end_date', ci.display_timezone) AS end_date,
+            access_rule->>'institution' AS institution
```

Comment:
Are this `AS` clauses doing anything here? My understanding is that in an `INSERT (...) SELECT (...)` it's the order of columns that matters for matching the selected columns to the inserted ones, and their names are irrelevant.

Reply:
Removed in e1e4645.